### PR TITLE
radio-button v1

### PR DESCRIPTION
**Pull request by abhikrishnaram**
image
image

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export { default as PageHeader } from './components/PageHeader';
 export { default as ProgressBar } from './components/ProgressBar';
 export { default as SkeletonItem } from './components/SkeletonItem';
 export { default as SimpleSelect } from './components/SimpleSelect';
+export { default as RadioGroup } from './components/Radio/RadioGroup';
```
Comment (Advaith3600) [inline src/index.ts:23]: Export single Radio Element as well

File: src/components/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          />
+          <span
+              className={clsx([
+                'dsr-inline-flex dsr-items-center dsr-justify-center dsr-flex-shrink-0',
+                'dsr-transition-shadow dsr-duration-normal dsr-border dsr-border-2 dsr-border-none dsr-rounded-full',
+                'dsr-border-inherit dsr-text-white', size?.button, isChecked && styled.radioButton,
+                isChecked ? color : isDarkTheme ? 'dsr-bg-white/20' : 'dsr-bg-gray-500/20',
+              ])}
+          />
+          <span className={clsx(['dsr-ml-2', size?.label && `dsr-text-${size?.label}`])}>{option.label}</span>
```
Comment (Advaith3600) [inline src/components/Radio/Radio.tsx:54]: `dsr-text-${size?.label}` - tailwind css JIT won't work with string interpolation, define the classes properly

Comment (Advaith3600) [reply src/components/Radio/Radio.tsx:54]: Accessibility: When you click on the label will it select the radio button? If not have to do it

File: src/components/Radio/Radio.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                'dsr-border-0 dsr-border-none dsr-h-px dsr-w-px dsr-p-0 dsr-whitespace-nowrap',
+                'dsr-overflow-hidden dsr-absolute -dsr-m-1',
+              ])}
+              style={{ clip: 'rect(0px, 0px, 0px, 0px)' }}
+          />
+          <span
+              className={clsx([
+                'dsr-inline-flex dsr-items-center dsr-justify-center dsr-flex-shrink-0',
+                'dsr-transition-shadow dsr-duration-normal dsr-border dsr-border-2 dsr-border-none dsr-rounded-full',
+                'dsr-border-inherit dsr-text-white', size?.button, isChecked && styled.radioButton,
```
Comment (Advaith3600) [inline src/components/Radio/Radio.tsx:50]: make it multiline

File: src/components/Radio/radio.module.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.radioButton {
+  &:before {
```
Comment (Advaith3600) [inline src/components/Radio/radio.module.scss:2]: `&::before` not `&:before`

**Review by Advaith3600 — CHANGES_REQUESTED** [review]

**Comment by Advaith3600** [comment]
Why does such an approach require such a simple design? 

You can use `accent-color` property of CSS for this and do absolute no madness

**Comment by abhikrishnaram** [comment]
> Why does such an approach require such a simple design?
> 
> You can use `accent-color` property of CSS for this and do absolute no madness

This approach uses a custom radio button instead of modifying the default HTML style. This approach I believe, gives us the ability to customize and modify completely as much as we require.

**Review by Advaith3600 — APPROVED** [review]

**Pull request merged**